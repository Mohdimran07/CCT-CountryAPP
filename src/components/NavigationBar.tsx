import { AppBar, Toolbar, Typography } from '@mui/material';
import React from 'react';

const NavigationBar = () => {
    return (
       <AppBar position='static' >
           <Toolbar>
               <Typography>Country Details</Typography>
           </Toolbar>
       </AppBar>
    );
};

export default NavigationBar;