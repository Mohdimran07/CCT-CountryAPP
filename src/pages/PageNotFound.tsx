import { Button } from '@mui/material';
import { Box } from '@mui/system';
import React from 'react';
import {useNavigate, NavigateFunction } from 'react-router-dom';

const PageNotFound = (): JSX.Element => {
    const navigate: NavigateFunction = useNavigate();
    const returnHandler = (): void => {
        navigate("/")
    }
    return (
        <Box>
            <h1>Page Not Found</h1>
            <Button variant='contained' onClick={returnHandler}>Back</Button>
        </Box>
    );
};

export default PageNotFound;