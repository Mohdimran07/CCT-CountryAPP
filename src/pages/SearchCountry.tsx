import { Button, FormControl } from "@mui/material";
import React, { ChangeEvent, FormEvent, useState } from "react";
import { NavigateFunction, useNavigate } from "react-router-dom";

const SearchCountry = (): JSX.Element => {
  const naviagte: NavigateFunction = useNavigate();
  const [countryName, setCountryName] = useState<string>("");
  const changeNameHandler = (e: ChangeEvent<HTMLInputElement>): void => {
    setCountryName(e.target.value);
  };

  const submitHandler = (e: FormEvent<HTMLFormElement>): void => {
    e.preventDefault();
    naviagte(`${countryName}`);
  };

  return (
    <FormControl
      component={"form"}
      onSubmit={submitHandler}
      sx={{ paddingTop: "20px" }}
    >
      <input
        type="text"
        value={countryName}
        onChange={changeNameHandler}
        placeholder="Enter country name"
        required
      ></input>{" "}
      <br />
      <Button type="submit" variant="contained">
        Search
      </Button>
    </FormControl>
  );
};

export default SearchCountry;
