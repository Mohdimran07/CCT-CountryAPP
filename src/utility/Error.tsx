import { Button } from '@mui/material';
import { Box } from '@mui/system';
import React from 'react';
import { NavigateFunction, useNavigate } from 'react-router-dom';

const Error = ():JSX.Element => {
    const navigate : NavigateFunction = useNavigate();
    const backButtonHandler = ():void => {
        navigate("/")
    }
    return (
       <Box>
           <h1>Something went wrong...</h1>
           <Button variant='contained' onClick={backButtonHandler}>Back</Button>
       </Box>
    );
};

export default Error;