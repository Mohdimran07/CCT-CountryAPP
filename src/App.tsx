import { Routes, Route } from "react-router-dom";
import "./App.css";
import NavigationBar from "./components/NavigationBar";
import CountryDetails from "./pages/CountryDetails";
import PageNotFound from "./pages/PageNotFound";
import SearchCountry from "./pages/SearchCountry";


function App(): JSX.Element {
  return <div className="App">
    <NavigationBar />
   <Routes>
     <Route path="/" element={<SearchCountry />}></Route>
     <Route path="/:name" element={<CountryDetails />} ></Route>
     <Route path="*" element={<PageNotFound />}></Route>
   </Routes>
    </div>;
}

export default App;
