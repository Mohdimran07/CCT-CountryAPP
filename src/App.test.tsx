import React from "react";
import { fireEvent, render, screen } from "@testing-library/react";
import App from "./App";
import { BrowserRouter } from "react-router-dom";
import SearchCountry from "./pages/SearchCountry";
import NavigationBar from "./components/NavigationBar";
import Error from "./utility/Error";
import PageNotFound from "./pages/PageNotFound"

describe("render App Component", () => {
  test("App Component", () => {
    const { container } = render(
      <BrowserRouter>
        <App></App>
      </BrowserRouter>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe("SearchCountry Component", () => {
  test("render the placeHoldertext", () => {
    render(
      <BrowserRouter>
      <SearchCountry />
      </BrowserRouter>
    );
    const placeHolderText = screen.getByPlaceholderText(/Enter country name/);
    expect(placeHolderText).toBeInTheDocument();
  })

  test("render the fireEvent placeHolder text", () => {
    render(
      <BrowserRouter>
      <SearchCountry />
      </BrowserRouter>
    );
    fireEvent.change(screen.getByPlaceholderText("Enter country name"), {
      target: { value: "India"}
    })
  })

  test("test for button", () => {
    render(
      <BrowserRouter>
      <SearchCountry />
      </BrowserRouter>
    );
     expect(screen.getByRole("button")).toBeInTheDocument();
  })
  test("render the no of buttons exist", () => {
    render(
      <BrowserRouter>
      <SearchCountry />
      </BrowserRouter>
    );
    expect(screen.getAllByRole("button").length).toBe(1);
  })

  test("render text box", () => {
    render(
      <BrowserRouter>
      <SearchCountry />
      </BrowserRouter>
    );
    expect(screen.getAllByRole("textbox").length).toBe(1);
  })
})

describe("render Naviagtion Bar component", () => {
  test("test for text", () => {
    render(
      <BrowserRouter>
      <NavigationBar />
      </BrowserRouter>
    )
    expect(screen.getByText(/Country/)).toBeInTheDocument();
  })

  test("test for banner", () => {
    render(
      <BrowserRouter>
      <NavigationBar />
      </BrowserRouter>
    )
    expect(screen.getAllByRole("banner").length).toBe(1)
  })
})

describe("render Error Component", () => {
  test("test for text element", () => {
    render(
      <BrowserRouter>
      <Error />
      </BrowserRouter>
    )
    expect(screen.getByText(/Something went wrong.../)).toBeInTheDocument();
  })
  test("test for button",() => {
    render(
      <BrowserRouter>
      <Error />
      </BrowserRouter>
    )
    expect(screen.getByRole("button")).toBeInTheDocument();
  })
  test("test for no of buttons", () => {
    render(
      <BrowserRouter>
      <Error />
      </BrowserRouter>
    )
    expect(screen.getAllByRole("button").length).toBe(1);
  })
  

})

describe("test on PageNotFound component", () => {
  test("test to finde existence any button", () => {
    render(
      <BrowserRouter>
        <PageNotFound></PageNotFound>
      </BrowserRouter>
    );
    const elemet = screen.getByRole("button");
    expect(elemet).toBeInTheDocument();
  });
});

describe("test on PageNotFound component", () => {
  test("test to find total number of buttons", () => {
    render(
      <BrowserRouter>
        <PageNotFound></PageNotFound>
      </BrowserRouter>
    );
    const elemet = screen.getAllByRole("button");
    expect(elemet.length).toBe(1);
  });

  test("finding text inside PageNotFound component", () => {
    render(
      <BrowserRouter>
        <PageNotFound></PageNotFound>
      </BrowserRouter>
    );

    const elemetText = screen.getByText(/Page Not Found/);
    expect(elemetText).toBeInTheDocument();
  });
});
