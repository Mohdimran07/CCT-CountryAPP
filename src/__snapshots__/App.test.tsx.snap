// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render App Component App Component 1`] = `
<div
  class="App"
>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
  >
    <div
      class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1t29gy6-MuiToolbar-root"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      >
        Country Details
      </p>
    </div>
  </header>
  <form
    class="MuiFormControl-root css-1ys1w97-MuiFormControl-root"
  >
    <input
      placeholder="Enter country name"
      required=""
      type="text"
      value=""
    />
     
    <br />
    <button
      class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="submit"
    >
      Search
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </form>
</div>
`;
